blocks/gallery/gallery.js

```diff
@@ -1,5 +1,68 @@
-console.log('gallery loaded');
+import { loadScript, loadCSS } from '../../scripts/aem.js';
+import { moveInstrumentation } from '../../scripts/scripts.js';
+
+await loadScript('https://cdn.jsdelivr.net/npm/@fancyapps/ui@5.0/dist/fancybox/fancybox.umd.js');
+await loadScript('https://cdn.jsdelivr.net/npm/@fancyapps/ui@5.0/dist/carousel/carousel.umd.js');
+await loadCSS('https://cdn.jsdelivr.net/npm/@fancyapps/ui@5.0/dist/fancybox/fancybox.css');
+await loadCSS('https://cdn.jsdelivr.net/npm/@fancyapps/ui@5.0/dist/carousel/carousel.css');
+/* global Fancybox, Carousel */
+
+let galleryCounter = 0;
 
 export default function decorate(block) {
-  console.log('gallery decorate fn', block);
+  galleryCounter += 1;
+
+  const slides = [...block.children].map((galleryItem) => {
+    const slideElement = document.createElement('div');
+    moveInstrumentation(galleryItem, slideElement);
+
+    const pictureElement = galleryItem.querySelector('picture');
+    const itemType = pictureElement ? 'gallery-item' : 'gallery-item-youtube';
+
+    let caption;
+    let fullscreenSrc;
+    let thumbnailSrc;
+
+    if (itemType === 'gallery-item') {
+      const captionElement = galleryItem.querySelector('p');
+      caption = captionElement.textContent;
+      const imgElement = pictureElement.querySelector('img');
+      const imgSrc = imgElement.getAttribute('src');
+      fullscreenSrc = imgSrc;
+      thumbnailSrc = imgSrc;
+    } else if (itemType === 'gallery-item-youtube') {
+      const [captionElement, youtubeIdElement] = [...galleryItem.querySelectorAll('p')];
+      caption = captionElement.textContent;
+      const videoID = youtubeIdElement.textContent;
+      fullscreenSrc = `https://www.youtube.com/watch?v=${videoID}`;
+      thumbnailSrc = `https://img.youtube.com/vi/${videoID}/maxresdefault.jpg`;
+    }
```

Comment:
@tamara027  This is a little brittle, because we have no way to target the specific sub items (`gallery-item` or `gallery-item-youtube`). Meaning that this would break if we define another submodel to the gallery which also does have the exact same structure but should be treated differently. Happy to have a call about that.

https://github.com/tamara027/fe-boilerplate/blob/main/component-definition.json#L169